+sw_tests/+unit_tests/unittest_spinw_optmagsteep.m

```diff
@@ -0,0 +1,219 @@
+            fields = {'datestart', 'dateend', 'obj', 'param'};
+            testCase.verify_val(rmfield(opt, fields), testCase.default_opt);
+            testCase.verify_obj(opt.obj, testCase.swobj)
+        end
+        
+        function test_converges_local_minimum_along_hard_axis(testCase)
+            % FM with S//a (hard axis) - no component along easy-axis
+            testCase.swobj.genmagstr('mode', 'direct', ...
+                                     'S', [1 1; 0 0; 0 0], ...
+                                     'k',[0,0,0], 'nExt', [2,1,1]);
+            testCase.swobj.optmagsteep;  % results in AFM S//a
+            expected_magstr = testCase.default_magstr;
+            expected_magstr.S = [-1, 1; 0, 0; 0, 0];
+            testCase.verify_val(testCase.swobj.magstr, expected_magstr);
+            testCase.verify_val(testCase.swobj.energy, -1.0)
+        end
+        
+        function test_spins_align_along_easy_axis_external_field(testCase)
+            testCase.swobj.field([0 0 1]);
+            % FM with S//a (hard axis) - no component along easy-axis
+            testCase.swobj.genmagstr('mode', 'direct', ...
+                                     'S', [1 1; 0 0; 0 0], ...
+                                     'k',[0,0,0], 'nExt', [2,1,1]);
+            testCase.swobj.optmagsteep('nRun', 150);  % results in AFM S//c
+            expected_magstr = testCase.default_magstr;
+            expected_magstr.S = [0, 0; 0, 0; 1, -1];
+            testCase.verify_val(testCase.swobj.magstr, expected_magstr, ...
+                'abs_tol', 1e-6);
+            testCase.verify_val(testCase.swobj.energy, -1.1, ...
+                'abs_tol', 1e-10)  % same energy as grd state without field
+        end
+        
+        function test_converges_global_minimum(testCase)
+            % FM with component of S // c (easy-axis)
+            testCase.swobj.genmagstr('mode', 'direct', ...
+                                     'S', [0 0; 1 1; 1 1], ...
+                                     'k',[0,0,0], 'nExt', [2,1,1]);
+            % try with one iterations - will not convergence
+            testCase.verifyWarning(...
+                @() testCase.swobj.optmagsteep('nRun', 1), ...
+                'spinw:optmagsteep:NotConverged');
+            testCase.swobj.optmagsteep('nRun', 150);  % results in AFM S//c
+            testCase.verify_val(testCase.swobj.magstr, ...
+                                testCase.default_magstr, 'abs_tol', 1e-8);
+            testCase.verify_val(testCase.swobj.energy, -1.1)
+        end
+        
+        function test_random_init_of_spins(testCase)
+            % FM with S//a (hard axis) - no component along easy-axis
+            % optmagsteep would converge at local not global minimum
+            testCase.swobj.genmagstr('mode', 'direct', ...
+                                     'S', [1 1; 0 0; 0 0], ...
+                                     'k',[0,0,0], 'nExt', [2,1,1]);
+            for irun = 1:5
```

Comment:
Yes sorry I definitely need a comment here to explain. The multiple runs is to account for the fact a random structure might lead to a local minimum. For example in the test `test_converges_local_minimum_along_hard_axis` - if the initial structure starts with spins aligned along the hard axis then it will not converge to the ground-state, there are probably other local minima it could get stuck in depending on the starting state.  I ran it 100 times and it always converged on the ground state, so I thought giving it 5 goes was probably sufficient!